### PR TITLE
Step 5: Machine learning with Scikit-learn

### PR DESCRIPTION
**Pull request by awilliamson1889**

### CODE REVIEW

File: email-classifier.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def data_set_cleaning(obj):
+    obj.to_lower()
+    obj.remove_digits()
+    obj.remove_punctuation_marks()
+    obj.remove_duplicates()
+    obj.remove_stopwords()
+
+url = "https://raw.githubusercontent.com/awilliamson1889/Python-DS-Learning/email-clf-model/tests/datasets/emails.csv"
```
Comment (jimmyhollywood) [inline email-classifier.py]: I would be better to use relative path here

File: Email-classification.ipynb [hunk 2]
```diff
@@ -0,0 +1,874 @@
+{
```
Comment (jimmyhollywood) [inline Email-classification.ipynb:1]: The notepad looks quite detailed - good job. I haven't ran it locally I will do it on next week but the information provided is quite good. The only thing I've found is the strange value of test data like ```with test size = 0.15%:```. Maybe you mean 15% there?

**Review by jimmyhollywood — APPROVED** [review]
I'd suggest you to configure one more job in GithubActions to check the accuracy of models by some threshold. Feel free to try different values.

**Comment by awilliamson1889** [comment]
<table border="1" class="dataframe">\n  <thead>\n    <tr style="text-align: right;">\n      <th></th>\n      <th>model</th>\n      <th>accuracy</th>\n      <th>precision</th>\n      <th>recall</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>3</th>\n      <td>MultinomialNB2 + tfidf</td>\n      <td>0.881</td>\n      <td>0.997</td>\n      <td>0.543</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>MultinomialNB1</td>\n      <td>0.963</td>\n      <td>0.961</td>\n      <td>0.897</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>SGDClassifier2 + tfidf</td>\n      <td>0.956</td>\n      <td>0.957</td>\n      <td>0.879</td>\n    </tr>\n    <tr>\n      <th>0</th>\n      <td>RandomForestClassifier1 + tfidf</td>\n      <td>0.952</td>\n      <td>0.94</td>\n      <td>0.88</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>RandomForestClassifier2 + chi</td>\n      <td>0.944</td>\n      <td>0.907</td>\n      <td>0.894</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>SGDClassifier1 + chi</td>\n      <td>0.954</td>\n      <td>0.903</td>\n      <td>0.937</td>\n    </tr>\n  </tbody>\n</table>

**Pull request closed without merging**